plugin/src/software/aws/toolkits/eclipse/amazonq/util/UpdateUtils.java

```diff
@@ -0,0 +1,123 @@
+
+    public static UpdateUtils getInstance() {
+        return INSTANCE;
+    }
+
+    private UpdateUtils() {
+        mostRecentNotificationVersion = Activator.getPluginStore().getObject(Constants.DO_NOT_SHOW_UPDATE_KEY, Version.class);
+        String localString = PluginClientMetadata.getInstance().getPluginVersion();
+        localVersion = ArtifactUtils.parseVersion(localString.substring(0, localString.lastIndexOf(".")));
+    }
+
+    private boolean newUpdateAvailable() {
+        //fetch artifact file containing version info from repo
+        remoteVersion = fetchRemoteArtifactVersion(REQUEST_URL);
+
+        //return early if either version is unavailable
+        if (remoteVersion == null || localVersion == null) {
+            return false;
+        }
+
+        //prompt should show if never previously displayed or remote version is greater
+        boolean shouldShowNotification = mostRecentNotificationVersion == null || remoteVersionIsGreater(remoteVersion, mostRecentNotificationVersion);
+
+        return remoteVersionIsGreater(remoteVersion, localVersion) && shouldShowNotification;
+    }
+
+    public void checkForUpdate() {
+        if (newUpdateAvailable()) {
+            showNotification();
+        }
+    }
+
+    private Version fetchRemoteArtifactVersion(final String repositoryUrl) {
+        HttpClient connection = HttpClientFactory.getInstance();
+        try {
+            HttpRequest request = HttpRequest.newBuilder()
+            .uri(URI.create(repositoryUrl))
+            .timeout(Duration.ofSeconds(5))
+            .GET()
+            .build();
+
+            HttpResponse<InputStream> response = connection.send(request,
+            HttpResponse.BodyHandlers.ofInputStream());
+
+            // handle response codes
+            if (response.statusCode() != HttpURLConnection.HTTP_OK) {
+                throw new AmazonQPluginException("HTTP request failed with response code: " + response.statusCode());
+            }
+
+            // process XZ content from input stream
+            try (InputStream inputStream = response.body();
+                 XZInputStream xzis = new XZInputStream(inputStream);
+                 BufferedReader reader = new BufferedReader(new InputStreamReader(xzis))) {
+
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    if (line.contains("<artifact classifier=\"osgi.bundle\"")) {
+                        int versionStart = line.indexOf("version=\"") + 9;
+                        int versionEnd = line.indexOf("\"", versionStart);
+                        String fullVersion = line.substring(versionStart, versionEnd);
+                        return ArtifactUtils.parseVersion(fullVersion.substring(0, fullVersion.lastIndexOf(".")));
+                    }
+                }
+            }
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
```

Comment:
We definitely should not affect other parts of the plugin if this fails, but we should at least log a message here.